[keyansheng] iP

keyansheng opened the pull request:
# Rubber Duke, a rubber duck for debugging 🦆 

> **If ducks are so smart, why don’t we just let the ducks do all the work?** It would be wonderful if this were true, but the fact is that most ducks prefer to take a mentoring role. - https://rubberduckdebugging.com/

Here are some things it can do:

- Quack
- Manage your tasks
- Help with your debugging sessions

To get started,

1. Download the latest [release](https://github.com/keyansheng/ip/releases)
1. `java -jar rubberduke.jar`
1. Add your tasks
1. Enjoy the mentorship of **Rubber Duke**, ~~a~~ *the* rubber duck

Here are some key features:

- [x] CLI
- [ ] GUI

Here is the entry point of the program:

```java
public static void main(String[] args) {
    new RubberDuke("./data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

keyansheng committed:
Rename bot to Rubber Duke

keyansheng committed:
Add greeting

keyansheng committed:
Add line break to greeting

keyansheng committed:
Add prompt and echo

keyansheng committed:
Add listing

keyansheng committed:
Add adding

keyansheng committed:
Add Task class

keyansheng committed:
Add marking and unmarking

keyansheng committed:
Add quacking to listing

keyansheng committed:
Add @Override to Task toString

keyansheng committed:
Add Task types

keyansheng committed:
Move command name trimming to main loop

This way, the strings passed to commands do not include the command
itself. This allows the logic to be more consistent between commands
that require specifying the command, and the default addTodo action
where the command name is optional.

I will keep the existing functionality of adding a todo by default if no
other command is specified, unless this turns out to be a bad idea as
the project evolves.

keyansheng committed:
Add adding different Task types

keyansheng committed:
Keep greeting within line length limit

keyansheng committed:
Add tests

keyansheng committed:
Add exception tests

keyansheng committed:
Add EmptyArgumentException

keyansheng committed:
Add UnknownCommandException

keyansheng committed:
Add EmptyArgumentException tests

keyansheng committed:
Add UnknownCommandException tests

keyansheng committed:
Add deleting

keyansheng committed:
Add deleting from empty list test

keyansheng committed:
Change commands to return strings instead of printing

keyansheng committed:
Add loading from file

keyansheng committed:
Change greetings and prompt from methods to constants

keyansheng committed:
Add saving to file

keyansheng committed:
Add loading and saving tests

keyansheng committed:
Merge branch 'branch-Level-7'

keyansheng committed:
Add date and time parsing

keyansheng committed:
Add date and time parsing tests

keyansheng committed:
Merge branch 'branch-Level-8'

keyansheng committed:
Change RubberDuke to non-static

keyansheng committed:
Change tasks from local variable to field

keyansheng committed:
Remove tasks from parameters

keyansheng committed:
Extract tasks to TaskList

keyansheng committed:
Change RubberDuke to private

keyansheng committed:
Clean up strings in TaskList

keyansheng committed:
Fix Event error messages

keyansheng committed:
Clean up strings in RubberDuke

keyansheng committed:
Change EmptyArgumentException to UserException

keyansheng committed:
Change UnknownCommandException to UserException

keyansheng committed:
Remove getCreateCommand from abstract Task

keyansheng committed:
Move Deadline parsing to deadline

keyansheng committed:
Move Event parsing to Event

keyansheng committed:
Move Todo parsing to Todo

keyansheng committed:
Extract file operations to Storage

keyansheng committed:
Extract parsing to Parser

keyansheng committed:
Remove duplicate prompt

keyansheng committed:
Change constants to inline

keyansheng committed:
Extract greetings to Ui

keyansheng committed:
Refactor RubberDuke to align with example

keyansheng committed:
Move showPrompt to Ui

keyansheng committed:
Refactor RubberDuke and Ui to align with example

keyansheng committed:
Use isExit

keyansheng committed:
Merge branch 'more-oop'

keyansheng committed:
Move everything to a rubberduke package

keyansheng committed:
Move tasks into rubberduke.tasks

keyansheng committed:
Move utils into rubberduke.utils

keyansheng committed:
Merge branch 'packages'

keyansheng committed:
Merge branch 'add-gradle-support'

keyansheng committed:
Add JUnit tests

keyansheng committed:
Merge branch 'junit'

keyansheng committed:
Create JAR package

keyansheng committed:
Merge branch 'jar'

keyansheng committed:
Add JavaDoc to Task

keyansheng committed:
Adhere to coding standard

keyansheng committed:
Add finding

keyansheng committed:
Merge branch 'branch-A-JavaDoc'

keyansheng committed:
Merge branch 'branch-A-CodingStandard'

keyansheng committed:
Merge branch 'branch-Level-9'

keyansheng committed:
Move ui assignment to field declaration

keyansheng committed:
Extract storage initialization to method

keyansheng committed:
Change storage to final

keyansheng committed:
Extract task loading to method

keyansheng committed:
Extract task saving to method

keyansheng committed:
Change from isExit back to break

keyansheng committed:
Handle EOF by exiting gracefully

keyansheng committed:
Add showing to Ui

showError uses System.out instead of System.err for now since tests
expect that.

keyansheng committed:
Extract command processing to method

keyansheng committed:
Extract input handling to method

keyansheng committed:
Extract file command processing to method

keyansheng committed:
Merge branch 'more-oop'

keyansheng committed:
Extract date time formatters to constant

keyansheng committed:
Extract date time formatting to method

keyansheng committed:
Merge branch 'more-oop'

keyansheng committed:
Rename packages from plural to singular

keyansheng committed:
Merge branch 'packages'

keyansheng committed:
Add Checkstyle

keyansheng committed:
Merge branch 'branch-A-CheckStyle'

keyansheng committed:
Fix import order

keyansheng committed:
Break before operator

keyansheng committed:
Add newline at end of file

keyansheng committed:
Merge branch 'codingstandard'

keyansheng committed:
Add javadoc to tasks

keyansheng committed:
Add javadoc to Parser

keyansheng committed:
Add javadoc to RubberDuke

keyansheng committed:
Add javadoc to Storage

keyansheng committed:
Add javadoc to UserException

keyansheng committed:
Add javadoc to Ui

keyansheng committed:
Add javadoc to TaskList

keyansheng committed:
Merge branch 'javadoc'

DanielJames0302 commented on src/main/java/rubberduke/task/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+ * Represents a deadline, which has an end date.
+ */
+public class Deadline extends Task {
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM);
```
Good naming of constant

DanielJames0302 commented on src/main/java/rubberduke/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+     *
+     * @return true if the task is done, false otherwise.
+     */
+    public boolean isDone() {
```
Good naming of boolean method

DanielJames0302 commented on src/main/java/rubberduke/util/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+     *
+     * @return string containing commands to recreate the current state of the task list.
+     */
+    public String dump() {
```
Perhaps more descriptive method name

DanielJames0302 commented on src/main/java/rubberduke/RubberDuke.java:
```diff
@@ -0,0 +1,83 @@
+        }
+    }
+
+    private void handleInput() {
```
Good method naming in this place and other places

DanielJames0302 reviewed (commented):
LGTM. Well done.

keyansheng committed:
Add JavaFX

keyansheng committed:
Add constructor with default file path

keyansheng committed:
Add images

keyansheng committed:
Add GUI

keyansheng committed:
Move Ui to ui package

keyansheng committed:
Extract Ui messages to constants

keyansheng committed:
Extract getResponse from processCommand

keyansheng committed:
Add response to MainWindow

keyansheng committed:
Add welcome and goodbye to GUI

keyansheng committed:
Add task saving to GUI

keyansheng committed:
Add CSS

keyansheng committed:
Add Launcher

keyansheng committed:
Set entry point to Launcher

keyansheng committed:
Flip text bubbles

keyansheng committed:
Merge branch 'branch-Level-10'

keyansheng committed:
Add assertions

keyansheng committed:
Improve code quality

keyansheng committed:
Extract methods in Storage

This adheres to SLAP and improves code quality.

keyansheng committed:
Sort imports

For consistency and code quality.

keyansheng committed:
Add javadoc

Checkstyle found missing javadoc in some files.

keyansheng committed:
Add friendlier command syntax

Single letter commands for creating new tasks and double letter commands
for managing tasks.

Double letter commands are inspired by unix (ls, rm) for familiarity to
the target audience who are CLI users.

keyansheng committed:
Merge pull request #3 from keyansheng/friendliersyntax

BCD-Extension C-FriendlierSyntax

keyansheng committed:
Merge branch 'master' into branch-A-CodeQuality

keyansheng committed:
Merge branch 'master' into branch-A-Assertions

keyansheng committed:
Merge pull request #2 from keyansheng/branch-A-CodeQuality

A-CodeQuality

keyansheng committed:
Merge branch 'master' into branch-A-Assertions

keyansheng committed:
Merge pull request #1 from keyansheng/branch-A-Assertions

A-Assertions

keyansheng committed:
Add a representative screenshot of the product

The file is in the docs folder and named Ui.png exactly.

Ui.png shows the full GUI window.

Ui.png is a single GUI window.

The product name is visible in the welcome message in the screenshot.

keyansheng committed:
Add user guide

keyansheng committed:
Update README

keyansheng committed:
Add quick start to user guide

keyansheng committed:
Remove quick start from README

keyansheng committed:
Update user guide

Fix links since GitHub Pages does not automatically detect links.

keyansheng committed:
Fix test method names

keyansheng committed:
Merge branch 'codingstandard'
